### PR TITLE
WT-8877 Always dump transaction state of active sessions on RTS failure

### PR DESCRIPTION
**Pull request by Sean04**
Removed the have diagnostic check to always dump active transaction information when rollback to stable fails due to active transactions. Also changed test_rollback_to_stable30 to reflect this change and to always check for the transaction state dump on failure.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -1400,13 +1400,11 @@ __rollback_to_stable_check(WT_SESSION_IMPL *session)
...
      */
     txn_active = __txn_user_active(session);
-#ifdef HAVE_DIAGNOSTIC
-    if (txn_active)
-        WT_TRET(__wt_verbose_dump_txn(session));
-#endif
 
-    if (txn_active)
+    if (txn_active){
+        WT_TRET(__wt_verbose_dump_txn(session));
```
Comment (etienneptl) [inline src/txn/txn_rollback_to_stable.c:1405]: Is it ok to dump this all the time? `__rollback_to_stable_check` is called at the beginning of `__rollback_to_stable`.
I can see in the original ticket we always want to print it, I guess it is fine then.
We could also do the following:
```suggestion
#ifdef HAVE_DIAGNOSTIC
  if(txn_active) {
#else
  if(txn_active && WT_VERBOSE_ISSET(session, WT_VERB_RTS)) {
#endif
    WT_TRET(__wt_verbose_dump_txn(session));
  }
```
However, this means we need the verbose category `WT_VERB_RTS` to be set when `HAVE_DIAGNOSTIC` is not enabled.

Comment (Sean04) [reply src/txn/txn_rollback_to_stable.c:1405]: Would that still help with the original goal of being able to diagnose BFs such as BF-24403 more easily? Whether it's okay or not I guess would also depend on how many times you might expect rollback to stable to fail due to active transactions?

Comment (etienneptl) [reply src/txn/txn_rollback_to_stable.c:1405]: It wouldn't be as much helpful as your changes. One nitpick though, I don't think we need `txn_active` anymore, we may just do `if(__txn_user_active(session))`.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Review by tammybailey — APPROVED** [review]
LGTM

**Pull request merged**